cmd/server/main.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"encoding/json"
+	"github.com/gorilla/mux"
+	"log"
+	"math/rand"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+)
+
+type product struct {
+	ID          int    `json:"id"`
+	Name        string `json:"name"`
+	Description string `json:"description"`
+	Price       int    `json:"price"`
+}
+
+var products []product
+
+func getProductsHandler(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	if err := json.NewEncoder(w).Encode(products); err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+	}
+}
+
+func initProducts() {
+	bs, err := os.ReadFile("products.json")
+	if err != nil {
+		log.Fatal(err)
+	}
+	if err := json.Unmarshal(bs, &products); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func main() {
+	initProducts()
+	r := mux.NewRouter()
+	r.HandleFunc("/products", getProductsHandler)
+
+	r.HandleFunc("/api/products", getProducts).Methods("GET")
```

Comment:
Test your server and ensure you can reach all the endpoints from a client. Order of routes matter.